src/model/misc.rs

```diff
@@ -53,6 +51,108 @@ impl_from_str! {
     ChannelId, ChannelIdParseError, parse_channel;
 }
 
+/// An image hash returned from the Discord API.
+///
+/// Note: This is parsed into a compact form when constructed, then turned back
+/// into the cannonical hex representation used by the API when needed.
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct ImageHash {
+    is_animated: bool,
+    hash: [u8; 16],
+}
+
+impl ImageHash {
+    #[must_use]
+    pub fn is_animated(&self) -> bool {
+        self.is_animated
+    }
+}
+
+impl std::fmt::Debug for ImageHash {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("\"")?;
+        <Self as std::fmt::Display>::fmt(self, f)?;
+        f.write_str("\"")
+    }
+}
+
+impl serde::Serialize for ImageHash {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.to_string().serialize(serializer)
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for ImageHash {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // TODO: Replace this with ArrayString<34>?
+        let helper = String::deserialize(deserializer)?;
+        Self::from_str(&helper).map_err(serde::de::Error::custom)
+    }
+}
+
+impl std::fmt::Display for ImageHash {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_animated {
+            f.write_str("a_")?;
+        }
+
+        for byte in self.hash {
+            write!(f, "{byte:02x}")?;
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ImageHashParseError {
```

Comment:
If `FromStr` will be the go-to method for constructing an `ImageHash`, the error type should be documented as well.